src/fu-engine.c

```diff
@@ -7096,6 +7096,216 @@ fu_engine_security_attrs_depsolve(FuEngine *self)
 }
 #endif
 
+/**
+ * fu_history_get_previous_security_attr:
+ * @self: a #FuHistory
+ * @appstream_id: maximum number of attributes to return, or 0 for no limit
+ * @current_setting: (nullable): current value
+ * @error: return location for a #GError, or %NULL
+ *
+ * Gets the security attributes of the previous BIOS setting for the given
+ * appstream ID and current BIOS config.
+ *
+ * Returns: (element-type #FuSecurityAttr) (transfer full): attr, or %NULL
+ *
+ * Since: 1.9.5
+ **/
+static FwupdSecurityAttr *
+fu_engine_get_previous_bios_security_attr(FuEngine *self,
+					  const gchar *appstream_id,
+					  const gchar *current_setting,
+					  GError **error)
+{
+	g_autoptr(GPtrArray) attrs_array = NULL;
+
+	attrs_array = fu_history_get_security_attrs(self->history, 20, error);
+	if (attrs_array == NULL)
+		return NULL;
+	for (guint i = 0; i < attrs_array->len; i++) {
+		FuSecurityAttrs *attrs = g_ptr_array_index(attrs_array, i);
+		g_autoptr(GPtrArray) attr_items = fu_security_attrs_get_all(attrs);
+		for (guint j = 0; j < attr_items->len; j++) {
+			FwupdSecurityAttr *attr = g_ptr_array_index(attr_items, j);
+			if (g_strcmp0(appstream_id, fwupd_security_attr_get_appstream_id(attr)) ==
+				0 &&
+			    g_strcmp0(current_setting,
+				      fwupd_security_attr_get_bios_setting_current_value(attr)) !=
+				0) {
+				g_debug("found previous BIOS setting for %s: %s",
+					appstream_id,
+					fwupd_security_attr_get_bios_setting_id(attr));
+				return g_object_ref(attr);
+			}
+		}
+	}
+
+	/* failed */
+	g_set_error_literal(error,
+			    G_IO_ERROR,
+			    G_IO_ERROR_NOT_SUPPORTED,
+			    "cannot find previous BIOS value");
+	return NULL;
+}
+
+/**
+ * fu_engine_fix_host_security_attr:
+ * @self: a #FuEngine
+ * @appstream_id: the Appstream ID
+ * @error: (nullable): optional return location for an error
+ *
+ * Fix one specific security attribute.
+ *
+ * Returns: %TRUE for success
```

Comment:
missing 1.9.6 in doc

Reply:
Nah, I think src stuff shouldn't have Since lines at all.